test/unit/helpers/account_helper_test.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class Api::AccountHelperTest < ActionView::TestCase
+  include AccountHelper
+
+  setup do
+    @provider = FactoryGirl.create(:simple_provider, provider_account: master_account)
+  end
+
+  attr_reader :provider
+  delegate :can?, to: :ability
+
+  class ProviderLoggedIn < Api::AccountHelperTest
+    def current_user
+      @current_user ||= FactoryGirl.create(:active_admin, account: provider)
+    end
+
+    def buyer_account
+      @buyer ||= FactoryGirl.create(:simple_buyer, provider_account: provider)
+    end
+
+    def delete_link_for(url, options = {})
+      assert_equal admin_buyers_account_path(buyer_account), url
+
+      confirm_option = options[:confirm]
+
+      assert_not_includes confirm_option, 'in 15 days'
+      assert_not_includes confirm_option, 15.days.from_now.to_date.to_s(:long)
+      assert_includes confirm_option, 'Are you sure you want to delete'
+    end
+  end
+
+  test 'delete_buyer_link' do
+    delete_buyer_link(buyer_account)
```

Comment:
I don't really like this way to test it because it is not obvious that the asserts are when this method reaches its last line and mocked in https://github.com/3scale/porta/pull/267/files#diff-616a76c4684185519c0a60f7e9646df9R24 and https://github.com/3scale/porta/pull/267/files#diff-616a76c4684185519c0a60f7e9646df9R57 . But I tried to do before this line something like:
```ruby
expects(:delete_link_for).with do |url, options|
  assert_arguments_delete_link_for(url, options)
end
```
and then rename in this file the 2 `delete_link_for` for `assert_arguments_delete_link_for`.
But it doesn't work because as soon as it does the mock for one of them, it seems to stay the same for the other one or something weird under the hood is going on...